File: dotnet/src/AgentDBException.cs
using System;

namespace Datacules.AgentDB
{
    /// <summary>
    /// The exception thrown when an AgentDB native operation fails.
    /// The <see cref="Exception.Message"/> property contains the text returned
    /// by <c>agentdb_last_error()</c> on the calling thread.
    /// </summary>
    public class AgentDBException : Exception
    {
        /// <inheritdoc />
        public AgentDBException(string message) : base(message) { }

        /// <inheritdoc />
        public AgentDBException(string message, Exception innerException)
            : base(message, innerException) { }
    }
}

File: java/src/main/java/com/datacules/agentdb/AgentDBException.java
package com.datacules.agentdb;

/**
 * Thrown when an AgentDB native operation fails.
 *
 * <p>The message is taken directly from {@code agentdb_last_error()} on the
 * calling thread, so it describes the actual native failure.
 */
public class AgentDBException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    public AgentDBException(String message) {
        super(message);
    }

    public AgentDBException(String message, Throwable cause) {
        super(message, cause);
    }
}
